README.md

```diff
@@ -1541,6 +1542,148 @@ expected that subject: /tmp/atrium-path/directory/subfolder/file        (sun.nio
+<a name="ex-because-3"></a>
+```text
+expected that subject: [1, 2]        (java.util.Arrays.ArrayList <1234789>)
+◆ is: empty
+```
+</ex-because-3>
+
+The same applies to null checks. For instance, instead of checking for null for an absent key in a map
+
+<ex-because-4>
+
+```kotlin
+expect(mapOf("a" to 1)["a"]).because("key a should not be in the map") {
+    toBe(null)
+}
+```
+↑ <sub>[Example](https://github.com/robstoll/atrium/tree/master/misc/tools/readme-examples/src/main/kotlin/readme/examples/ReadmeSpec.kt#L386)</sub> ↓ <sub>[Output](#ex-because-4)</sub>
+<a name="ex-because-4"></a>
+```text
+expected that subject: 1        (kotlin.Int <1234789>)
+◆ equals: null
+ℹ because: key a should not be in the map
+```
+</ex-because-4>
+
+Write
+
+<ex-because-5>
+
+```kotlin
+expect(mapOf("a" to 1)).containsNotKey("a")
+```
+↑ <sub>[Example](https://github.com/robstoll/atrium/tree/master/misc/tools/readme-examples/src/main/kotlin/readme/examples/ReadmeSpec.kt#L391)</sub> ↓ <sub>[Output](#ex-because-5)</sub>
+<a name="ex-because-5"></a>
+```text
+expected that subject: {a=1}        (java.util.Collections.SingletonMap <1234789>)
+◆ does not contain key: "a"        <1234789>
+```
+</ex-because-5>
+
+Or in case of a nullable property:
+
+<ex-because-6>
+
+```kotlin
+expect(IllegalArgumentException("no no").message)
+    .because("it should result in an exception without message") {
+        toBe(null)
+    }
+```
+↑ <sub>[Example](https://github.com/robstoll/atrium/tree/master/misc/tools/readme-examples/src/main/kotlin/readme/examples/ReadmeSpec.kt#L395)</sub> ↓ <sub>[Output](#ex-because-6)</sub>
+<a name="ex-because-6"></a>
+```text
+expected that subject: "no no"        <1234789>
+◆ equals: null
+ℹ because: it should result in an exception without message
+```
+</ex-because-6>
+
+Use a feature assertion, which includes it in reporting, already giving the context needed.
+
+<ex-because-7>
+
+```kotlin
+expect(IllegalArgumentException("no no")).feature { f(it::message) }.toBe(null)
+```
+↑ <sub>[Example](https://github.com/robstoll/atrium/tree/master/misc/tools/readme-examples/src/main/kotlin/readme/examples/ReadmeSpec.kt#L402)</sub> ↓ <sub>[Output](#ex-because-7)</sub>
+<a name="ex-because-7"></a>
+```text
+expected that subject: java.lang.IllegalArgumentException
+◆ ▶ message: "no no"        <1234789>
+    ◾ equals: null
+```
+</ex-because-7>
+
+Following a checklist as a rule of thumb:
+1. check if there is not already a predefined assertion function
+2. use a [feature assertion](#feature-assertions) instead.
+3. write an own assertion function (e.g. a [boolean based Assertions](#boolean-based-assertions)).
+4. [let us know](https://github.com/robstoll/atrium/issues/new?template=feature_request.md&title=using because) why you still want to use `because` - maybe Atrium lacks another feature
+5. use `because`
+
+</details>
+
```

Comment:
Okay, so I’ll rewrite my comment about this block. You’ll have to imagine that the initial one was a lot more eloquent :laughing: 

I don’t like this block in its current form. I think I know where you are coming from: You don’t want users to degenerate Atrium into a boolean matching function. I get the thought. Somebody coming from the omnipresent JUnit who is used to writing 

```Java
assertFalse(list.isEmpty(), "found no matching elements!")
```

might indeed get the idea to write 

```
expect(list.isEmpty()).because("found no matching elements") { toBe(false) }
```

_However_, I think there is a strong reason for every user not to write something like that: _The rest of the README_. I can’t imagine somebody learning about `because` before having seen how to use Atrium correctly before. I also don’t think that anybody would voluntarily write something as ugly as that, once they have learned how to do it correctly.

With that in mind, I see the following disadvantages with the README block in its current form:

 1. One should _never_ give negative examples. Never. First, study after study has shown that negative examples can often lead to the opposite effect of what was intended. Second, people scan through READMEs and copy the first block of code they see, without reading the surrounding text.
 2. The block is quite long for something that could be summarized as ‘please use Atrium according to the very README you are currently reading’
 3. I think the block makes it sound like `because` is a last resort. But it isn’t. `because` is for documenting non-obvious requirements. This use case clearly distinguishable from building a missing expectation function.
 4. I have a feeling that it is not easy to understand what the block is trying to tell us if one is not reading carfully.


I’d like us to try to reduce this block to a handful of sentences. Something among the lines of:

> 💬  Use `because` only to give reasons for non-obvious expectations.

> `because` can be a useful tool for explaining why there is a certain expectation. Sometimes, it is not directly obvious why we should expect something. In such cases, using `because` can make your code and your error messages easier to understand for other developers (including yorself in three months).

> Having said that, you should _not_ use `because` if you are missing a specific predefined expectation function. You can use [feature extractors](#feature-assertions), write [your own expectation function](#boolean-based-assertions) or [propose an addition to Atrium](https://github.com/robstoll/atrium/issues/new?template=feature_request.md&title=Missing%20Expectation%20Function) in such cases. 

> Just like code comments, `because` can be helpful, but should not be overused.



I have the strong feeling that my wording was a lot better in lost comment … :woman_shrugging:

Reply:
@jGleitz note that this information is inside a details block which one needs to expand first (see https://github.com/robstoll/atrium/tree/%23660-because#attach-a-reason). So most users will not read it anyway... especially not the ones skimming through the README to copy & paste code. But I am fine with changing this to your text. I figured it is more convincing if you see the results but yeah... in the end I don't really care if people I am not working with are using it differently than I would. Hence we can also delete the whole block. What do you think?

Reply:
I'm aware that it is in a details block, however, this address only my first point, if at all. So I think changing it will be an improvement 👍

Reply:
OK, if I get you right, then you vote for keeping the details block but improve it. right?

Reply:
@jGleitz In both cases (keep and improve or delete), I let you do the adjustments and in the same go, you can improve the samples in api-fluent and api-infix :wink:

Reply:
>  OK, if I get you right, then you vote for keeping the details block but improve it. right?

Exactly.

> I let you do the adjustments

will do!